veniq/dataset_collection/augmentation.py

```diff
@@ -0,0 +1,287 @@
+    for method_node in dict_method_lines:
+        if invoked_method_node.member == method_node.name:
+            return method_node
+    return None
+
+
+def _is_match_to_the_conditions(method_invoked: ASTNode) -> bool:
+    if method_invoked.parent.node_type == ASTNodeType.THIS:
+        parent = method_invoked.parent.parent
+        class_names = [x for x in method_invoked.parent.children if hasattr(x, 'string')]
+        member_references = [x for x in method_invoked.parent.children if hasattr(x, 'member')]
+        children = [x for x in member_references if x.member != method_invoked.member] + class_names
+    else:
+        parent = method_invoked.parent
+        children = True
+
+    is_not_parent_member_ref = not (method_invoked.parent.node_type == ASTNodeType.MEMBER_REFERENCE)
+    is_not_chain_before = not (parent.node_type == ASTNodeType.METHOD_INVOCATION) and not children
+    chains_after = [x for x in method_invoked.children if x.node_type == ASTNodeType.METHOD_INVOCATION]
+    is_not_chain_after = not chains_after
+    is_not_inside_if = not (parent.node_type == ASTNodeType.IF_STATEMENT)
+    is_not_inside_while = not (parent.node_type == ASTNodeType.WHILE_STATEMENT)
+    is_not_inside_for = not (parent.node_type == ASTNodeType.FOR_STATEMENT)
+    is_not_enhanced_for_control = not (parent.node_type == ASTNodeType.ENHANCED_FOR_CONTROL)
+    # ignore case else if (getServiceInterface() != null) {
+    is_not_binary_operation = not (parent.node_type == ASTNodeType.BINARY_OPERATION)
+    is_not_ternary = not (parent.node_type == ASTNodeType.TERNARY_EXPRESSION)
+    # if a parameter is any expression, we ignore it,
+    # since it is difficult to extract with AST
+    is_actual_parameter_simple = all([hasattr(x, 'member') for x in method_invoked.arguments])
+    is_not_class_creator = not (parent.node_type == ASTNodeType.CLASS_CREATOR)
+    is_not_cast = not (parent.node_type == ASTNodeType.CAST)
+    is_not_array_creator = not (parent.node_type == ASTNodeType.ARRAY_CREATOR)
+    is_not_lambda = not (parent.node_type == ASTNodeType.LAMBDA_EXPRESSION)
+    other_requirements = all([
+        is_not_chain_before,
+        is_actual_parameter_simple,
+        is_not_chain_after,
+        is_not_inside_if,
+        is_not_inside_while,
+        is_not_binary_operation,
+        is_not_ternary,
+        is_not_class_creator,
+        is_not_cast,
+        is_not_array_creator,
+        is_not_parent_member_ref,
+        is_not_inside_for,
+        is_not_enhanced_for_control,
+        is_not_lambda,
+        not method_invoked.arguments])
+
+    if (not method_invoked.qualifier and other_requirements) or \
+            (method_invoked.qualifier == 'this' and other_requirements):
+        return True
+    else:
+        return False
+
+
+def determine_type(method_node: ASTNode):
+    """
+
+    :param method_node: Method declaration
+    :return: int - type
+    """
+
+    if not method_node.parameters:
+        return 0
+    else:
```

Comment:
delete 'else return'

Reply:
i will use here more complex structure, since it's not all code I am going to use. In future I will write the full function for type identification